Update Schemas API to include format keyword (#3333)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/api/schemas/index.md

| Property                       | Description                                | DataType                                          | Nullable | Unique | Readonly |
|:-------------------------------|:-------------------------------------------|:--------------------------------------------------|:---------|:-------|:---------|

> | title                          | user-defined display name for the property | String                                            | FALSE    | FALSE  | FALSE    |

Author: Sure, np. There are numerous tables with this format; I probably just followed existing style. Let me fix them all up.

| type                           | type of property                           | `string`, `boolean`, `number`, `integer`, `array` | FALSE    | TRUE   | TRUE     |
| enum                           | enumerated value of the property           | array                                             | TRUE     | TRUE   | FALSE    |
| oneOf                          | non-empty array of valid JSON schemas      | array                                             | TRUE     | TRUE   | FALSE    |
| format                         | identifies the type of data represented by the string | string                                            | TRUE     | TRUE   | FALSE    |

##### Description details

- `enum`: The value of the property is limited to one of the values specified in the `enum` definition. The list of values for the `enum` must be made up of unique elements.

- `oneOf`: Okta only supports `oneOf` for specifying display names for an `enum`. Each schema has the following format:

  ```json
  {
    "const": "enumValue",
    "title": "display name"
  }
  ```

>   In case `enum` is used in conjunction with `oneOf`, the set of enumerated values and their order must be kept.

Reviewer: ```suggestion
  When `enum` is used in conjunction with `oneOf`, you must keep the set of enumerated values and their order.
```

  `oneOf` is only supported in conjunction with `enum`, providing a mechanism to return a display name for the `enum` value.

- `format`: Okta uses this keyword to identify the type of data represented by the string. The following attribute formats are supported: `uri`, `date-time`, `email`, `ref-id`, `encrypted`, `hashed`, `country-code`, `language-code`, `locale`, and `timezone`. See the following request that uses the format property:

  ```bash

>   curl --location --request POST 'https://{$yourOktaDomain}/api/v1/meta/schemas/apps/0oa4cthzyqgFj8ISJ5d7/default' \

Reviewer 2: The endpoint is wrong for user schemas. Should be something like
```
POST 'https://${yourOktaDomain}/api/v1/meta/schemas/user/default'
```
or
```
POST 'https://${yourOktaDomain}/api/v1/meta/schemas/user/${typeId}`
```

| type                           | type of property                           | `string`, `boolean`, `number`, `integer`, `array` | FALSE    | FALSE  | FALSE    |
| enum                           | enumerated value of the property           | array                                             | TRUE     | TRUE   | FALSE    |
| oneOf                          | non-empty array of valid JSON schemas      | array                                             | TRUE     | TRUE   | FALSE    |

> | format                         | identifies the type of date represented by the string | string                                            | TRUE     | TRUE   | FALSE    |

Author: Thanks. Good catch.

| format                         | validates the format type of the property      | string                                            | TRUE     | TRUE   | FALSE    |

##### Description Details

- `enum`: The value of the property is limited to one of the values specified in the `enum` definition. The list of values for the `enum` must be made up of unique elements.

- `oneOf`: Okta only supports `oneOf` for specifying display names for an `enum`. Each schema has the following format:

  ```json
  {
    "const": "enumValue",
    "title": "display name"
  }
  ```

  In case `enum` is used in conjunction with `oneOf`, the set of enumerated values and their order must be kept.

  ```json
  {"enum": ["S","M","L","XL"],
    "oneOf": [
      {"const": "S", "title": "Small"},
      {"const": "M", "title": "Medium"},
      {"const": "L", "title": "Large"},
      {"const": "XL", "title": "Extra Large"}
    ]
  }
  ```

  `oneOf` is only supported in conjunction with `enum`, providing a mechanism to return a display name for the `enum` value.

> - `format`: Okta supports the `format` keyword and validates the following format attributes: `uri`, `date-time`, `email`, `ref-id`, `encrypted`, `hashed`, `country-code`, `language-code`, `locale`, and `timezone`. See the following request that uses the format property:

Author: Hi @benzhao It was included in the JIRA ticket from support.

> - `format`: Okta supports the `format` keyword and validates the following format attributes: `uri`, `date-time`, `email`, `ref-id`, `encrypted`, `hashed`, `country-code`, `language-code`, `locale`, and `timezone`. See the following request that uses the format property:

Reviewer 2: I don't think we should add documentation that we support this because I don't think there's validation for most, if not all of this in custom schemas. I recall there was a bug for country-code not being validated in custom schemas, but it looks like that was reverted for whatever reason.

| type                           | type of property                           | `string`, `boolean`, `number`, `integer`, `array` | FALSE    | FALSE  | FALSE    |
| enum                           | enumerated value of the property           | array                                             | TRUE     | TRUE   | FALSE    |
| oneOf                          | non-empty array of valid JSON schemas      | array                                             | TRUE     | TRUE   | FALSE    |

> | format                         | identifies the type of data represented by the string | string                                            | TRUE     | TRUE   | FALSE    |

Reviewer: the other descriptions in that column should start with a capital letter, also.
```suggestion
| format                         | Identifies the type of data represented by the string | string                                            | TRUE     | TRUE   | FALSE    |
```

##### Description details

- `enum`: The value of the property is limited to one of the values specified in the `enum` definition. The list of values for the `enum` must be made up of unique elements.

- `oneOf`: Okta only supports `oneOf` for specifying display names for an `enum`. Each schema has the following format:

  ```json
  {
    "const": "enumValue",
    "title": "display name"
  }
  ```

>   In case `enum` is used in conjunction with `oneOf`, the set of enumerated values and their order must be kept.

Reviewer: ```suggestion
  When `enum` is used in conjunction with `oneOf`, you must keep the set of enumerated values and their order.
```

- `enum`: The value of the property is limited to one of the values specified in the `enum` definition. The list of values for the `enum` must be made up of unique elements.

- `oneOf`: Okta only supports `oneOf` for specifying display names for an `enum`. Each schema has the following format:

  ```json
  {
    "const": "enumValue",
    "title": "display name"
  }
  ```

  In case `enum` is used in conjunction with `oneOf`, the set of enumerated values and their order must be kept.

  ```json
  {"enum": ["S","M","L","XL"],
    "oneOf": [
      {"const": "S", "title": "Small"},
      {"const": "M", "title": "Medium"},
      {"const": "L", "title": "Large"},
      {"const": "XL", "title": "Extra Large"}
    ]
  }
  ```

  `oneOf` is only supported in conjunction with `enum`, providing a mechanism to return a display name for the `enum` value.

- `format`: Okta uses this keyword to identify the type of data represented by the string. The following attribute formats are supported: `uri`, `date-time`, `email`, `ref-id`, `encrypted`, `hashed`, `country-code`, `language-code`, `locale`, and `timezone`. See the following request that uses the format property:

  ```bash

>   curl --location --request POST 'https://{$yourOktaDomain}/api/v1/meta/schemas/apps/0oa4cthzyqgFj8ISJ5d7/default' \

Reviewer 2: Consider replacing `0oa4cthzyqgFj8ISJ5d7` with a placeholder like `${appId}`

Also, looks like `{$yourOktaDomain}` should be `${yourOktaDomain}`
